torch/quantization/fx/_equalize.py

```diff
@@ -1,11 +1,13 @@
 import torch
-import torch.nn as nn
-from torch.quantization.observer import MinMaxObserver, PerChannelMinMaxObserver
+from torch.quantization.observer import (
+    MinMaxObserver, PerChannelMinMaxObserver, ObserverBase
+)
+from torch.quantization.qconfig import QConfig
 
 import warnings
 
 
-class _InputEqualizationObserver(nn.Module):
+class _InputEqualizationObserver(ObserverBase):
```

Comment:
I'm not super familiar with the equalization code, but are we sure this is the right abstraction to have this inherit from `ObserverBase`?  It seems unexpected that an instance of `ObserverBase` would then have children that are also instances of `ObserverBase`.  Thoughts?

Reply:
I originally had it as an `nn.Module` but with the new changes, in order to create a default QConfig, I needed a `with_args` function which is located in `ObserverBase`. @jerryzh168 suggested to have the class inherit `ObserverBase` which made sense since this class functions as an Observer. Conversely, another solution which I had before could be to just import `_with_args()` and create a `with_args` function separately in `_InputEqualizationObserver` and `_WeightEqualizationObserver`.

Reply:
IMO might be worth considering not having a class have children of the same base class, it might be hard for code readers to understand.  I'd vote for sticking with `nn.Module` as base class.